### ISSUE TITLE
`ptr_arg` false negative when `ImplIterator` is used

### ISSUE THREAD

**Issue by Alexendoo** [rust-lang/rust-clippy#8995]
### Summary

When the `IntoIterator` impl of a `&Vec` is used the lint doesn't fire

### Lint Name

ptr_arg

### Reproducer

```rust
fn f(x: &Vec<u32>) {
    for _ in x {}
}
```

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=045036269cc6a1a1399da4154c1a65d1)

**Comment by FishyBusinessYT** [comment]
Are there plans to fix this? Is there help needed?
